modules/aruco/src/aruco.cpp

```diff
@@ -1563,7 +1563,7 @@ void refineDetectedMarkers(InputArray _image, const Ptr<Board> &_board,
 
         _detectedCorners.create((int)finalAcceptedCorners.size(), 1, CV_32FC2);
         for(unsigned int i = 0; i < finalAcceptedCorners.size(); i++) {
-            _detectedCorners.create(4, 1, CV_32FC2, i, true);
+            _detectedCorners.create(1, 4, CV_32FC2, i, true);
```

Comment:
Why?

Below we have 4,1 again: `_rejectedCorners.create(4, 1, ...)`

Reply:
It is surprising that the dimension of the matrices `out[i], _objPoints[i], _detectedCorners[i], _rejectedCorners[i], _diamondCorners[i]` is: (1 (rows) , 4 (cols)). But `create` with (4 (rows), 1 (cols)) was used.
![why did m's size change?](https://user-images.githubusercontent.com/22337800/142664467-8d02c220-f0ac-4871-825e-d63e2132ca24.png)
Flag `allowTransposed` is ignored in this case (same values for true/false).
Looks like a bug in `void _OutputArray::create(int _rows, int _cols, int mtype, int i, bool allowTransposed, int fixedDepthMask) const`.

Reply:
Such changes affecting API are considered as a "breaking change".

So proper and complete change should have:
- clear justification
- consistent solution (fix 1 place only and left 6 others is not a good way)
- adapt coding rules to avoid such/similar problems in the future.

---

> surprising

There is `allowTransposed` parameter on this line.

Reply:
This strange code change is removed. Now `refineDetectedMarkers()` for create `OutputArrayOfArrays corners` and `rejectedCorners` uses `_copyVector2Output()`, as `detectMarkers()`.

It's strange that before `detectMarkers()` and `refineDetectedMarkers()` created `OutputArrayOfArrays corners` and `rejectedCorners` in different ways.